### ISSUE TITLE
Allow the Rust plugin to build only certain targets, specifically to support testing

### ISSUE THREAD

**Issue by timothyfroehlich** [google/automotive-design-compose#191]
The Rust plugin is a bit blunt and always compiles all targets, and these builds can't run in parallel (they all get triggered in parallel, but cargo puts in a lock file that makes the other builds wait until the lock is cleared). Right now we're compiling four targets and that's only necessary for release. 

It would speed up build time if only the required target were built. I think the best way to do it would be to set up alternative build flavors for different targets, a debugX86_64 target and so-on. For testing on an emulator you'd build the version that matches your emulator's architecture.

This will make the biggest difference when working on the Rust code itself, and it can reduce CI time for running emulator tests (once we get that running again). It'll also reduce the size of the APKs that need to be uploaded/installed/etc.

Next steps on this would be to do some build analysis to see the actual speed-up from just building one target.

**Comment by timothyfroehlich** [comment]
@rylin8 @iamralpht @srahman80 Thoughts?

**Comment by rylin8** [comment]
Sounds like a good idea, but I would put it at a fairly low priority. At least for me, I don't have any issues with build times at the moment.

**Comment by timothyfroehlich** [comment]
Ok, sounds good. Let me know if it starts to be a problem and you'd like me to move forward on it.
